### PR TITLE
5141: Hapi request measurer implementation and spec

### PR DESCRIPTION
**Pull request by FBajo**

### CODE REVIEW

File: spec/components/hapi/hapiRequestMeasurerSpec.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  const requestMock = {} as hapi.Request & any;
+
+  beforeEach(() => {
+    serverMock.ext = () => void 0;
+    optionsMock.statsdClient = statsDMock;
+    hMock.continue = Symbol('continue');
+    requestMock.app = {};
+    requestMock.method = 'GET';
+    requestMock.path = '/v0/product-lists/social-lists';
+    requestMock.response = <hapi.ResponseObject>{statusCode: 200};
```
Comment (GWeggemans) [inline spec/components/hapi/hapiRequestMeasurerSpec.ts]: Use `as  hapi.ResponseObject` for casting, maybe we can force this with tslint?

https://palantir.github.io/tslint/rules/no-angle-bracket-type-assertion/

File: src/components/hapi/hapiRequestMeasurer.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
+import * as hapi from 'hapi';
+import Boom from 'boom';
+import {StatsD} from 'hot-shots';
+
+declare module 'hapi' {
+  interface ApplicationState {
+    statsdClient: StatsD;
```
[the remark this answers is not in the document]
Comment (FBajo) [reply src/components/hapi/hapiRequestMeasurer.ts]: Misunderstood our call I guess. Removed

File: src/components/hapi/hapiRequestMeasurer.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    'request.duration',
+    Date.now() - request.info.received,
+    extractStatsDTagsFromRequest(request));
+
+  return h.continue;
+};
+
+const lifecycleRegistration = (server: hapi.Server, options: HapiRequestMeasurerOptions) => {
+  server.ext('onRequest', createStatsdClientLifecycleMethod(options));
+  server.ext('onPreResponse', statsdTimingLifecycleMethod());
```
[the remark this answers is not in the document]
Comment (FBajo) [reply src/components/hapi/hapiRequestMeasurer.ts]: Exactly

File: spec/components/hapi/hapiRequestMeasurerSpec.ts (hunk 1 above)
Comment (FBajo) [reply spec/components/hapi/hapiRequestMeasurerSpec.ts]: Updated and added the rule

**Pull request merged**